Purchases/NSError+RCExtensions.m

```diff
@@ -0,0 +1,37 @@
+//
+// Created by Andrés Boedo on 2/24/20.
+// Copyright (c) 2020 Purchases. All rights reserved.
+//
+
+#import "NSError+RCExtensions.h"
+#import "RCPurchasesErrors.h"
+#import "RCBackend.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+
+@implementation NSError (RCExtensions)
+
+- (BOOL)shouldMarkSyncedKeyPresent {
+    BOOL isNetworkError = self.code == RCNetworkError;
+    BOOL shouldMarkSynced = (
+        !isNetworkError
+        && self.userInfo[RCShouldMarkSyncedKey] != nil
+        && ((NSNumber *) self.userInfo[RCShouldMarkSyncedKey]).boolValue
+    );
+    if (shouldMarkSynced) {
+        return YES;
+    } else if (self.userInfo[NSUnderlyingErrorKey]) {
+        NSError *underlyingError = (NSError *) self.userInfo[NSUnderlyingErrorKey];
+        if (underlyingError) {
+            return underlyingError.shouldMarkSyncedKeyPresent;
+        }
+    }
+
+    return NO;
+}
```

Comment:
I am a little bit confused about what's going on here. Can't you just check for  `RCShouldMarkSyncedKey` in the userInfo? For the network error there won't be any `RCShouldMarkSyncedKey`, so you can just do `if [self.userInfo[RCShouldMarkSyncedKey]) boolValue]`, since if it doesn't exist, it will evaluate to false. I would need to test that, but look into how it's done when posting the receipts RCPurchases:857.

I am also confused about the recursive call in line 27. I don't think the underlying errors will ever contain the should mark as synced key right?

Reply:
I'll double-check the exact behavior, but the reason I added the recursive call in the first place was because the original error was getting wrapped inside another one, so a shallow search would bring a false negative

Reply:
Was this fine then?

Reply:
just re-tested, and you're right, this isn't necessary. The value in the user info comes in for the top-most error, not the underlying one. 
I think I might have gotten confused while testing for network errors ¯\_(ツ)_/¯
In any case, I killed the recursive call and simplified the checks